packages/expo-notifications/src/NotificationsEmitter.ts

```diff
@@ -0,0 +1,73 @@
+import { EventEmitter, Subscription, NativeModulesProxy } from '@unimodules/core';
+import { Notification, NotificationResponse } from './NotificationsEmitter.types';
+
+interface NotificationReceivedEvent {
+  type: 'notificationReceived';
+  notification: Notification;
+}
+
+interface NotificationResponseEvent {
+  type: 'notificationResponseReceived';
+  response: NotificationResponse;
+}
+
+interface NotificationsDroppedEvent {
+  type: 'notificationsDropped';
+}
+
+export type NotificationEvent =
+  | NotificationReceivedEvent
+  | NotificationResponseEvent
+  | NotificationsDroppedEvent;
+
+export type NotificationListener = (notification: NotificationEvent) => void;
+
+// Web uses SyntheticEventEmitter
+const emitter = new EventEmitter(NativeModulesProxy.ExpoNotificationsEmitter);
+const didReceiveNotificationEventName = 'onDidReceiveNotification';
+const didDropNotificationsEventName = 'onNotificationsDeleted';
+const didReceiveNotificationResponseEventName = 'onDidReceiveNotificationResponse';
+
+type CompoundSubscription = Subscription & {
+  __subscriptions: Subscription[];
+};
+
+export function addNotificationListener(listener: NotificationListener): Subscription {
```

Comment:
what do you think about exposing separate listener functions for these three types of events? It seems like the ways in which you would handle each event would be fairly disjoint and so it's maybe clearer to just have three different events you subscribe to with different listeners.

Reply:
I was afraid of polluting the namespace with too many functions, but now I think you're right that the listeners will be disjoint and it makes sense to split the emitter functionality.

Following your suggestion I will land this PR now and make the changes on a separate branch. 👍